bigchaindb/block.py

```diff
@@ -229,3 +232,89 @@ def _start(self):
         p_write.start()
         p_delete.start()
 
+
+class BacklogDeleteRevert(Block):
+
+    def __init__(self, q_backlog_delete):
+        # invalid transactions can stay deleted
+        self.q_tx_to_validate = q_backlog_delete
+        self.q_tx_validated = mp.Queue()
+        self.q_transaction_to_revert = mp.Queue()
+        self.q_tx_delete = mp.Queue()
+
+        self.monitor = Monitor()
+
+    def locate_transactions(self):
+        """
+        Determine if a deleted transaction has made it into a block
+        """
+        # create bigchain instance
+        b = Bigchain()
+
+        while True:
+            tx = self.q_tx_validated.get()
+
+            # poison pill
+            if tx == 'stop':
+                self.q_tx_delete.put('stop')
+                self.q_transaction_to_revert.put('stop')
+                return
+
+            # check if tx is in a (valid) block
+            validity = b.get_blocks_status_containing_tx(tx['id'])
+
+            if validity and list(validity.values()).count(Bigchain.BLOCK_VALID) == 1:
```

Comment:
Do we need to check the validity of the block? This seems to be doing overlapping work with #193 

The way I see it there are 3 different possibilities:
1. if a transaction was deleted but made it into a block regardless of validity, its ok (if the block is later invalid #193 should take care of the rest).
2. If the transaction did not made it into a block and is invalid, its ok
3. If the transaction did not made it into a block and is valid, revert